persistence-cassandra-3.11/src/main/java/org/apache/cassandra/metrics/CassandraMetricsRegistry.java

```diff
@@ -20,51 +25,75 @@
  * <p>This class must mimic the exact public API of the original one.
  */
 public class CassandraMetricsRegistry extends MetricRegistry {
```

Comment:
This class is copied from the original Cassandra source code, right?
Is there also a test class for it? (`CassandraMetricsRegistryTest`). If yes, then I think we should port it to the Stargate database, (and adapt to the new JMX changes)

Reply:
`CassandraMetricsRegistryTest` indeed exists, but at first glace it does not look like it covers all the registration paths, just some of them. Plus if the code evolves on the C* side and the test is updated we are still not going to get the updates automatically, so the risk of behaviour divergence remains.

Reply:
Yes, It does not solve the problem of `behaviour divergence`, only lack of testing